### PR TITLE
Run the Desktop app in a sandbox

### PR DESCRIPTION
**Pull request by dbkr**
* Turn off node integration in the electron renderer process
 * Enable the chromium sandbox to put the renderer into its own process
 * Expose just the ipc module with a preload script
 * Introduce a little IPC call wrapper so we can call into the
   renderer process and await on the result.
 * Use this in a bunch of places we previously used direct calls
   to electron modules.
 * Convert other uses of node, eg. use of process to derive the
   platform (just look at the user agent)
 * Strip out the desktopCapturer integration which doesn't appear
   to have ever worked (probably best to just wait until
   getDisplayMedia() is available in chrome at this point:
   https://github.com/vector-im/riot-web/issues/4880).

Requires https://github.com/matrix-org/matrix-react-sdk/pull/2372

### CODE REVIEW

File: src/vector/index.js [hunk 1]
```diff
@@ -219,8 +221,15 @@ async function loadApp() {
     const fragparts = parseQsFromFragment(window.location);
     const params = parseQs(window.location);
 
-    // set the platform for react sdk (our Platform object automatically picks the right one)
-    PlatformPeg.set(new Platform());
+    // set the platform for react sdk
+    //if (navigator.userAgent.toLowerCase().indexOf('electron') > 0) {
```
Comment (jryans) [inline src/vector/index.js]: Maybe remove this commented code?

File: webpack.config.js [hunk 2]
```diff
@@ -108,11 +108,6 @@ module.exports = {
             "matrix-js-sdk": path.resolve('./node_modules/matrix-js-sdk'),
         },
     },
-    externals: {
```
Comment (jryans) [inline webpack.config.js]: I am not sure I follow why this was removed... It seems unrelated to the rest of the changes. Is it something outdated that's no longer needed?

Comment (MTRNord) [reply webpack.config.js]: The webpack side doesn't import electron anymore from what I can see. Means that it doesn't need to be declared anymore in the webpack config. (webpack side == everything that actually is riot)
